packages/react/src/input.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from "@hpcc-js/preact-shim";
+
+export interface IInput {
+    type: "number" | "range" | "date" | "text" | "select" | "checkbox";
+    value: number | string | boolean;
+    className?: string;
+    attributes?: any;
+    callbacks?: any;
+}
+
+export const Input: React.FunctionComponent<IInput> = ({
+    type = "text",
+    className,
+    value,
+
+    attributes = {}
+}) => {
+    return <input
+        value={value}
+        {...attributes}
```

Comment:
attributes should be specific and not general (will help re-use in the long run)

Reply:
Okay, I was on the fence about this too... more vs less explicit. What are the right attributes for input here?

At first I had something like this:
```
const Input = {
    // attributes: value, placeholder, className
    //    onkeyup, onclick, onfocus, onblur, oninput
}
const NumericInput = {
    // type="number" | "range"
    // attributes: min, max, step
}
const StringInput = {
    // type="text"
    // no unique attributes come to mind
}
const BooleanInput = {
    // type="checkbox"
    // attributes: checked
}
```

Reply:
@GordonSmith does the above look good?

Reply:
I don't think you want to have NumericInput with types of number + range.  Rather you would have NumericInput and RangeInput?